### PR TITLE
🚧 Add suggestion models

### LINKED ISSUE #100: Create the Autocomplete suggestion type
## Description
We need a discriminated union to represent all possible autocomplete suggestions that can appear for the user. Since the view is new, we won't copy the [Xamarin implementation](https://github.com/toggl/mobileapp/tree/develop/Toggl.Core/Autocomplete/Suggestions) 1:1. The possible cases are:

- Time Entry Suggestion
- Project Suggestion
- Task Suggestion
- Tag Suggestion
- Create Project Suggestion
- Create Tag Suggestion

## Definition of done
- [ ] The type is added to our codebase

### PR DESCRIPTION
**Pull request by sofokli1**
Closes #100 

not sure about the last 2 points in the issue (`Create Project Suggestion`, `Create Tag Suggestion`) so didn't do anything for those.

### CODE REVIEW

**Comment by heytherewill** [comment]
I'm not sure we need the protocol that's basically a marker interface. We need a proper discriminated union (in Swift's case, an enum) so that we are sure to always handle all possible cases

**Comment by heytherewill** [comment]
Regarding the entity creation ones, the first option in the list of suggestions is `create project XXX`, so we need to consider that as well

File: Modules/Models/Models/Types/AutocompleteSuggestionType.swift [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import Foundation
+
+public enum AutocompleteSuggestionType {
+    case timeEntrySuggestion
```
Comment (heytherewill) [inline Modules/Models/Models/Types/AutocompleteSuggestionType.swift:4]: Each case should also hold the metadata needed for that entry to be created. For a timeEntrySuggestion, it needs to hold a time entry. For tasks, a project and a task. For the createProject, a string with the project name

**Comment by heytherewill** [comment]
Droid implementation for reference https://github.com/toggl/android/pull/167/files

File: Modules/Models/Models/Types/AutocompleteSuggestionType.swift [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 public enum AutocompleteSuggestionType {
-    case timeEntrySuggestion
-    case projectSuggestion
-    case taskSuggestion
-    case tagSuggestion
-    case createProjectSuggestion
-    case createTagSuggestion
+    case timeEntrySuggestion(timeEntry: TimeEntry)
+    case projectSuggestion(project: Project)
+    case taskSuggestion(task: Task)
```
Comment (heytherewill) [inline Modules/Models/Models/Types/AutocompleteSuggestionType.swift:6]: The task suggestion also need the project data

Comment (sofokli1) [reply Modules/Models/Models/Types/AutocompleteSuggestionType.swift:6]: It is hard to think what this cases need without actually having to use these

**Pull request merged**